endpoint.go

```diff
@@ -46,13 +49,29 @@ type EndpointOption func(ep *endpoint)
 // ContainerData is a set of data returned when a container joins an endpoint.
 type ContainerData struct {
 	SandboxKey string
-	HostsPath  string
 }
 
 type containerConfig struct {
-	hostName   string
-	domainName string
-	generic    map[string]interface{}
+	hostName       string
+	domainName     string
+	generic        map[string]interface{}
+	hostsPath      string
+	ExtraHosts     []extraHost
```

Comment:
shouldnt this be `extraHosts` and can you please add a comment for a use-case that requires such a support.